src/main/java/application/models/User.java

```diff
@@ -12,7 +12,7 @@
     private String password;
     private String username;
 
-    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @OneToMany(mappedBy = "reviewAuthor", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
```

Comment:
So I think to avoid the duplicate review entry issue, we can change the cascade type here (or in products) to avoid the Cascade on persist (though we should change the cascade on delete). I'm not sure for which we should change the cascade type, we'll have to make sure that we're always persisting to the one we keep it in.